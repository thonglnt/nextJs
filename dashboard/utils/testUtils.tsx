import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
