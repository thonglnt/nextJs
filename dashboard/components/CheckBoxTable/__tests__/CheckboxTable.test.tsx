// Utils
import { MOCK_DATA_CUSTOMERS } from '@constants/mockData'
import { fireEvent, render, screen } from '@utils/testUtils'

// Components
import TableCheckbox, { ITableCheckbox } from '..'

describe('Checkbox Table component', () => {
  const mockProps: ITableCheckbox = {
    rows: MOCK_DATA_CUSTOMERS,
  }

  it('should render Checkbox Table component', () => {
    const { container } = render(<TableCheckbox {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should active cell after click button checkbox', () => {
    const { container } = render(<TableCheckbox {...mockProps} />)

    const checkbox = container.querySelectorAll(
      "input[type='checkbox']",
    )[0] as HTMLInputElement
    fireEvent.click(checkbox)

    expect(checkbox.checked).toBe(true)
  })

  it('should render next page after click button next page on table', () => {
    render(<TableCheckbox {...mockProps} />)

    const button = screen.getByRole('button', {
      name: /go to next page/i,
    })

    fireEvent.click(button)

    const viewPagination = screen.getByText(/6â€“10 of 10/i)

    expect(viewPagination).toBeInTheDocument()
  })
})
