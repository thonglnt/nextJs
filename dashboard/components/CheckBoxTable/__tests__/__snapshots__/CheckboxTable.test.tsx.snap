// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Table component should render Checkbox Table component 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          aria-labelledby="tableTitle"
          class="MuiTable-root css-1gplja-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-wvwcdf-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-hbtnrh-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-16rj92e-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-label="select all desserts"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-1242m0a-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-1242m0a-MuiTableCell-root"
                scope="col"
              >
                Location
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeMedium css-pcfucs-MuiTableCell-root"
                scope="col"
              >
                orders
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeMedium css-pcfucs-MuiTableCell-root"
                scope="col"
              >
                Spent
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeMedium css-pcfucs-MuiTableCell-root"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              aria-checked="false"
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-1nf681w-MuiTableCell-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-16rj92e-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-0"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone MuiTableCell-sizeMedium css-12jl0j-MuiTableCell-root"
                id="enhanced-table-checkbox-0"
                scope="row"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png"
                  />
                </div>
                <div
                  class="MuiBox-root css-7pf6at"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                    >
                      Ekaterina Tankova
                    </h5>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                    >
                      carson.darrin@devias.io
                    </p>
                  </div>
                </div>
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-11w94w9-MuiTableCell-root"
              >
                Parkersburg, West Virginia, USA
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                300
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="MoreVertIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              aria-checked="false"
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-1nf681w-MuiTableCell-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-16rj92e-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-1"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone MuiTableCell-sizeMedium css-12jl0j-MuiTableCell-root"
                id="enhanced-table-checkbox-1"
                scope="row"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png"
                  />
                </div>
                <div
                  class="MuiBox-root css-7pf6at"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                    >
                      Cao Yu
                    </h5>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                    >
                      carson.darrin@devias.io
                    </p>
                  </div>
                </div>
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-11w94w9-MuiTableCell-root"
              >
                Iowa, Bristow, USA
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                300
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="MoreVertIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              aria-checked="false"
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-1nf681w-MuiTableCell-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-16rj92e-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-2"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone MuiTableCell-sizeMedium css-12jl0j-MuiTableCell-root"
                id="enhanced-table-checkbox-2"
                scope="row"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png"
                  />
                </div>
                <div
                  class="MuiBox-root css-7pf6at"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                    >
                      Alexa Richardson
                    </h5>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                    >
                      carson.darrin@devias.io
                    </p>
                  </div>
                </div>
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-11w94w9-MuiTableCell-root"
              >
                Atlanta, Georgia, USA
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                300
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="MoreVertIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              aria-checked="false"
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-1nf681w-MuiTableCell-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-16rj92e-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-3"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone MuiTableCell-sizeMedium css-12jl0j-MuiTableCell-root"
                id="enhanced-table-checkbox-3"
                scope="row"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png"
                  />
                </div>
                <div
                  class="MuiBox-root css-7pf6at"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                    >
                      Anje Keizer
                    </h5>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                    >
                      carson.darrin@devias.io
                    </p>
                  </div>
                </div>
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-11w94w9-MuiTableCell-root"
              >
                Dover, Ohio, USA
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                300
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="MoreVertIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              aria-checked="false"
              class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-1nf681w-MuiTableCell-root"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-root MuiCheckbox-colorSecondary css-16rj92e-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-4"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </td>
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone MuiTableCell-sizeMedium css-12jl0j-MuiTableCell-root"
                id="enhanced-table-checkbox-4"
                scope="row"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="https://material-kit-pro-react.devias.io/assets/avatars/avatar-jie-yan-song.png"
                  />
                </div>
                <div
                  class="MuiBox-root css-7pf6at"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1arc1mz-MuiTypography-root"
                    >
                      Clarke Gillebert
                    </h5>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-10ca3tv-MuiTypography-root"
                    >
                      carson.darrin@devias.io
                    </p>
                  </div>
                </div>
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-paddingNone MuiTableCell-sizeMedium css-11w94w9-MuiTableCell-root"
              >
                Dallas, Texas, USA
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                300
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="MoreVertIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":r1:"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r1: :r0:"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r0:"
              role="button"
              tabindex="0"
            >
              5
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="5"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            1–5 of 10
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
